Sprint 9 solution http api

AlexanderTuzhikov opened the pull request.

AlexanderTuzhikov committed:
refactor: методы создания задач, обновления и удаления. Обновлена структура контроля пересечений по времени

AlexanderTuzhikov committed:
feat: реализовать HTTP API для TaskManager

- Добавлен класс HttpTaskServer для запуска HTTP-сервера (порт 8080).
- Подключена библиотека Gson для работы с JSON.
- Реализованы обработчики (handlers) для пяти основных путей:

/task - работа с обычными задачами
/subtask - работа с подзадачами
/epic - работа с эпиками
/history - получение истории просмотра
/prioritized -получение списка задач по приоритету

- Для каждого обработчика добавлен маппинг методов TaskManager:

GET - чтение
POST - добавление/обновление
DELETE - удаление

- Написаны unit-тесты

AlexanderTuzhikov committed:
fix: subtasksForEpic.computeIfAbsent(subtask.getEpicId(), _ -> new ArrayList<>())

maks-avilow commented on test/ru/yandex/practicum/server/EpicHttpHandlerTest.java:
```diff
@@ -0,0 +1,185 @@
+package ru.yandex.practicum.server;
+
```
папку test нужно пометить как test source root. делается это правой кнопкой мыши выбрав mark directory as ->

maks-avilow commented on src/ru/yandex/practicum/manager/impl/FileBackedTaskManager.java:
```diff
@@ -29,51 +26,61 @@ public FileBackedTaskManager(Path saveFile) {
-                        fileBackedTaskManager.updateSubtaskTime((Subtask) task);
-
-                        int epicId = subtask.getEpicId();
+                        fileBackedTaskManager.putSubtask((Subtask) task);
```
Принцип инкапсуляции (Encapsulation)
«Внутренние детали реализации должны быть скрыты» 

Логика загрузки задач раскрыта прямо в switch
Клиент (в данном случае — сам FileBackedTaskManager) знает слишком много о том, как обрабатываются разные типы задач
Исправление: инкапсулировать детали реализации в приватные методы
Тогда switch будет просто маршрутизировать задачу, а не выполнять логику

maks-avilow commented on src/ru/yandex/practicum/Main.java:
```diff
@@ -1,87 +1,30 @@
-                epicId);
-        taskManager.setTimeTask(subtask, LocalDateTime.of(2025, 8, 13, 10, 0),
-                Duration.ofMinutes(30));
+    public static void main(String[] args) throws IOException {
```
Exception 'java.io.IOException' is never thrown in the method  это значит что throws IOException не должно здесь быть. это нарушает принцип yagni.  (You Aren't Gonna Need It) — «Ты этого не будешь использовать».

maks-avilow reviewed (changes requested):
Привет! давай доработаем немного

AlexanderTuzhikov committed:
refactor: улучшения структуры и инкапсуляции

- Пометил папку test как Test Source Root для корректного распознавания тестов в IDE
- Принцип инкапсуляции: детали загрузки задач из файла вынесены в приватные методы (loadTask, loadSubtask, loadEpic, updateSubtaskFromEpic, updateLoadId)
- Удалён ненужный throws IOException

maks-avilow reviewed (approved):
принято
